src/shared/modules.mjs

```diff
@@ -854,6 +854,167 @@ const generateEventListenResponse = json => {
     return json
 }
 
+const getRefDefinition = (uri = '', moduleJson = {}) => {
+  if (!uri) {
+    throw "getRefDefinition requires a non-null uri parameter"
+  }
+  if (uri.startsWith('#')) {
+    const path = (uri).split('#').pop()
+
+    let definition
+    if (path) {
+     definition = getPathOr(null, path.slice(1).split('/'), moduleJson)
+    }
+    if (!definition) {
+      throw `getRefDefinition: Path '${uri}' not found in ${moduleJson ? (moduleJson.title || moduleJson.info.title) : moduleJson}.`
+      return null
+    }
+    else {
+      return definition
+    }
+  }
+  else {
+    throw `Cannot call getRefDefinition with ${uri}`
+  }
+}
+
+const getAnyOfSchema = (inType, json, anyOfTypes) => {
+    let outType = localizeDependencies(inType, json)
+    if (outType.schema.anyOf) {
+        if (anyOfTypes.length > 0) {
+        }
+        else {
+            let definition = ''
+            if (inType.schema['$ref'] && (inType.schema['$ref'][0] === '#')) {
+                definition = getRefDefinition(inType.schema['$ref'], json, json['x-schemas'])
+            }
+            else {
+                definition = outType.schema
+            }
+            definition.anyOf.forEach(anyOf => {
+                anyOfTypes.push(anyOf)
+            })
+            outType.schema = anyOfTypes
+        }
+    }
+    return outType
+}
+
+const generateAnyOfSchema = (anyOf, name, summary) => {
+    let anyOfType = []
+    if (!anyOf['$ref']) {
+        anyOf.name = anyOf.title || anyOf.name || ''
+        delete anyOf.title
+        if (anyOf.schema === undefined) {
+            anyOf.schema = {}
+            if (anyOf.oneOf) {
+                anyOf.schema.oneOf = anyOf.oneOf
+                delete anyOf.oneOf
+            }
+            else if (anyOf.anyOf) {
+                anyOf.schema.anyOf = anyOf.anyOf
+                delete anyOf.anyOf
+            }
+            else if (anyOf.allOf) {
+                anyOf.schema.allOf = anyOf.allOf
+                delete anyOf.allOf
+            }
+        }
+    }
+    anyOfType.name = name[0].toLowerCase() + name.substr(1)
+    anyOfType.summary = summary
+    anyOfType.schema = anyOf
+    return anyOfType
+}
+
+const generateParamsAnyOfSchema = (methodParams, anyOf, anyOfTypes, title, summary) => {
+    let params = []
+    methodParams.forEach(p => {
+        if (p.schema === anyOfTypes) {
+            let anyOfType = generateAnyOfSchema(anyOf, title, summary)
+            anyOfType.required = p.required
+            params.push(anyOfType)
+        }
+        else {
+            params.push(p)
+        }
+    })
+    return params
+}
+
+const generateResultAnyOfSchema = (method, methodResult, anyOf, anyOfTypes, title, summary) => {
+    let methodResultSchema = Object.assign({}, method.result.schema)
```

Comment:
I'm not sure this does a deep copy... so you could be introducing side effects when you modify this object. Not sure if that's desired or not, but we try to avoid it in individual methods, especially in the /shared/ folder.